### PR TITLE
Added Cruise Control System Tests

### PR DESCRIPTION
**Pull request by tomncooper**
### Type of change

Enhancement / new feature

### Description

- Added deployment with CC to Kafka ST resources
- Added pod name and deployment resources
- Added CC deployment system test

### Checklist

- [X] Write tests
- [X] Make sure all tests pass
- [X] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java [hunk 1]
```diff
@@ -203,4 +203,15 @@ public static String zookeeperServiceName(String clusterName) {
...
+    /**
+     * Returns the name of the Cruise Control {@code Deployment} for a {@code Kafka} cluster of the given name.
+     * This {@code Deployment} will only exist if {@code Kafka.spec.cruiseControl} is configured in the
+     * {@code Kafka} resource with the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Cruise Control {@code Deployment}.
+            */
+    public static String cruiseControlDeploymentName(String clusterName) {
+        return clusterName + "-cruise-control";
+    }
```
Comment (kyguy) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java]: The deployment name can be grabbed from the CruiseControlResources [1]

[1] https://github.com/kyguy/strimzi-kafka-operator/blob/cruise-control-deployment/api/src/main/java/io/strimzi/api/kafka/model/CruiseControlResources.java#L13-L20

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java [hunk 2]
```diff
@@ -223,12 +229,18 @@ private static Kafka waitFor(Kafka kafka) {
...
+        // Kafka Exporter is not setup every time
         if (kafka.getSpec().getKafkaExporter() != null) {
             LOGGER.info("Waiting for Kafka Exporter pods");
             DeploymentUtils.waitForDeploymentReady(KafkaExporterResources.deploymentName(name));
             LOGGER.info("Kafka Exporter pods are ready");
         }
+        // Cruise Control is not setup every time
+        if (kafka.getSpec().getCruiseControl() != null) {
+            LOGGER.info("Waiting for Cruise Control pods");
+            DeploymentUtils.waitForDeploymentReady(KafkaResources.cruiseControlDeploymentName(name));
```
Comment (kyguy) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java]: The deployment name can be grabbed from the CruiseControlResources [1]

[1] https://github.com/kyguy/strimzi-kafka-operator/blob/cruise-control-deployment/api/src/main/java/io/strimzi/api/kafka/model/CruiseControlResources.java#L13-L20

**Review by kyguy — APPROVED** [review]
Excellent work @tomncooper. The tests run smoothly on my machine. This will be really nice to have as we add more features

File: systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        KafkaResource.kafkaWithCruiseControl(CLUSTER_NAME, 3, 3).done();
+
+        String ccStatusCommand = "curl -X GET localhost:9090/kafkacruisecontrol/state";
+        String ccPodName = PodUtils.getFirstPodNameContaining("cruise-control");
+        LOGGER.info("Using pod: " + ccPodName);
+        String zkResult = cmdKubeClient().execInPodContainer(ccPodName, "cruise-control",
+                "/bin/bash", "-c", ccStatusCommand).out();
+
+        assertThat(zkResult, not(containsString("404")));
+        assertThat(zkResult, containsString("RUNNING"));
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java]: zkResult? is it just a typo?

**Review by ppatierno — APPROVED** [review]
LGTM other than comments from Kyle and one nit a left.

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java (hunk 1 above)
Comment (tomncooper) [reply api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java]: Oh right, missed that class. I will switch to that.

File: systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java (hunk 3 above)
Comment (tomncooper) [reply systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java]: Yep that is due to me spending too much time with ZooKeeper!

**Comment by tomncooper** [comment]
You guys happy to merge this?

**Comment by ppatierno** [comment]
@tomncooper +1

**Pull request merged**